### PR TITLE
(fix) Fixes Carthage support for tvOS

### PR DESCRIPTION
**Pull request by robinmacharg**
## Goal

[#523](https://github.com/bugsnag/bugsnag-cocoa/issues/523) notes an issue when using Carthage to build a project utilising Bugsnag that was causing tvOS builds to attempt to use the iOS project and thereby fail to build.  This PR addresses this issue.

## Design

The XCode build setting `SUPPORTED_PLATFORMS` was blank in the iOS project implying using the base SDK and masking the correct project for building tvOS.  Carthage, in turn, then used the iOS project to build the tvOS framework.  By setting this property to `iOS` Carthage then correctly finds the `tvOS` project.

## Tests

The Travic CI Makefile has been updated to also now run the Carthage tvOS build where previously it was only building MacOS/OSX and iOS.

### Outstanding Questions

- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review
  - [X] Release

- [ ] The correct target branch has been selected (`master` for fixes, `next` for
  features)
- [ ] If this is intended for release have all of the [pre-release checks](CONTRIBUTING.md) been considered?
- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [X] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [X] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: iOS/Bugsnag.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1514,6 +1518,7 @@
 				);
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SKIP_INSTALL = YES;
+				SUPPORTS_MACCATALYST = YES;
```
Comment (tomlongridge) [inline iOS/Bugsnag.xcodeproj/project.pbxproj:1521]: Is this change required?

Comment (robinmacharg) [reply iOS/Bugsnag.xcodeproj/project.pbxproj:1521]: Yes.  This change allows the CI system (which runs MacOS 10.14 - without Catalyst support) to (trivially) find and replace the line with `SUPPORTS_MACCATALYST = NO;` and allow the static library to build under CI conditions.  A manual test of a Catalyst-compatible static library build on 10.15 is also performed during UAT.

**Comment by fractalwrench** [comment]
Closing as this should be addressed by #670 instead
